Added ContentPicker Validation For Required Fields

Sandeep905 opened the pull request:
Fixes https://github.com/OrchardCMS/OrchardCore/issues/17462

I tried to fix the above-linked Issue. @Piedone  I am not sure if this approach is acceptable or not. If not let me know how I can fix the issue. This will be the first-ever open-source contribution from my end. 
![image](https://github.com/user-attachments/assets/1521751f-d529-4b62-a32f-0358c5cec0c9)

hishamco commented on src/OrchardCore.Modules/OrchardCore.ContentFields/Views/ContentPickerField.Edit.cshtml:
```diff
@@ -12,6 +12,7 @@
     var searchUrl = Url.RouteUrl(new { area = "OrchardCore.ContentFields", controller = "ContentPickerAdmin", action = "SearchContentItems", part = partName, field = fieldName });
     var vueElementId = $"ContentPicker_{partName}_{fieldName}_{Guid.NewGuid().ToString("n")}";
     var multiple = settings.Multiple.ToString().ToLowerInvariant();
+    bool IsValidationRequired = settings.Required && !Model.SelectedItems.Any() ? true : false;
```
```suggestion
    var IsValidationRequired = settings.Required && !Model.SelectedItems.Any();
```

hishamco commented on src/OrchardCore.Modules/OrchardCore.ContentFields/Views/ContentPickerField.Edit.cshtml:
```diff
@@ -42,7 +43,7 @@
             </ul>
 
-            <div class="w-lg-75 w-xxl-50">
+            <div class="w-lg-75 w-xxl-50 @(IsValidationRequired ? "Content-Picker-Required":string.Empty)">
```
```suggestion
            <div class="w-lg-75 w-xxl-50 @(IsValidationRequired ? "Content-Picker-Required" : string.Empty)">
```

hishamco reviewed (changes requested):
Regarding the style don't change this file instead change the one in the `Styles` folder

https://github.com/OrchardCMS/OrchardCore/blob/3de1e7ff9e28a9b4afa1d25abe77bdd21e661ad2/src/OrchardCore.Modules/OrchardCore.Resources/ResourceManagementOptionsConfiguration.cs#L493-L498

Piedone commented on src/OrchardCore.Modules/OrchardCore.ContentFields/Views/ContentPickerField.Edit.cshtml:
```diff
@@ -42,7 +43,7 @@
             </ul>
 
-            <div class="w-lg-75 w-xxl-50">
+            <div class="w-lg-75 w-xxl-50 @(IsValidationRequired ? "Content-Picker-Required" : string.Empty)">
```
This displays the validation also when opening an editor, still before saving. Better would be to behave like any other validation, and only show after save (publish).

Sandeep905 replied:
Thanks for the feedback i will fix the issues and update you.

Sandeep905 replied:
![image](https://github.com/user-attachments/assets/10360cc8-471b-4ff3-91a5-d619488ae4f0)
@Piedone  We can check the model state along with the current check so it will work only on user submission not on the first load. Is that a good idea?

Piedone replied:
Yep, that's a good idea.

Piedone reviewed (changes requested):
Thank you!

Yep, what Hisham says. See https://docs.orchardcore.net/en/latest/guides/gulp-pipeline/. That's why https://github.com/OrchardCMS/OrchardCore/actions/runs/13251045412?pr=17470 fails.

When addressing review feedback, please adhere to the following:

- Please update your pull request according to feedback until it is approved by one of the core team members.
- Apply [suggested changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes) directly so the reviewer doesn't have to eyeball the changes. These resolve themselves after applying them, and that's fine.
- Don't resolve other conversations so it's easier to track for the reviewer. Then, the reviewer will resolve them.
- Feel free to mark conversations that you addressed to keep track of them with an emoji or otherwise, just don't resolve them.
- Please keep conversations happening in line comments in those convos, otherwise, communication will be a mess. If you have trouble finding them, see [this video](https://github.com/OrchardCMS/OrchardCore/pull/14749#issuecomment-1917976028).
- When you're done addressing all feedback of a review, click ["Re-request review"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#re-requesting-a-review) in the top-right corner for each reviewer when you're ready for another round of review, so they know that you're done.

Sandeep905 commented:
> Regarding the style don't change this file instead change the one in the `Styles` folder
> 
> https://github.com/OrchardCMS/OrchardCore/blob/3de1e7ff9e28a9b4afa1d25abe77bdd21e661ad2/src/OrchardCore.Modules/OrchardCore.Resources/ResourceManagementOptionsConfiguration.cs#L493-L498

Unable to find the Style folder in OrchardCore.Resources -> Style 
> vue-multiselect.min.css is only located in 
![image](https://github.com/user-attachments/assets/ec6c2bd6-5ec1-4d5e-b599-ab92eb09ed3b)

Let me know what am I missing @hishamco  ?

hishamco commented:
It seems it's a bug that I need to fix after I check the input for the assets coming from `node_modules/vue-multiselect/dist/vue-multiselect.min.css`, so I suggest creating`vue-multiselect-override.css`, so we don't need to touch the distributed file. I remembered I did something similar in the past

Your thoughts @Piedone?

hishamco commented:
No bug here, the asset file will output the styles & scripts in the `wwwroot`

Piedone commented:
You'll either need to add a new SCSS file to `OrchardCore.ContentFields/Assets
/scss` and include it (like the others are), or better yet, check where the red border comes from for the other fields and add that class in the code.

hishamco commented:
@Sandeep905 could you please share the style that has been added for this PR, looking into a minified version is hard, then we can check how we can go

Sandeep905 commented:
`.Content-Picker-Required .multiselect__tags{
    border:1px solid red! important;
    background-image:url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 12 12' width='12' height='12' fill='none' stroke='%23dc3545'%3e%3ccircle cx='6' cy='6' r='4.5'/%3e%3cpath stroke-linejoin='round' d='M5.8 3.6h.4L6 6.5z'/%3e%3ccircle cx='6' cy='8.2' r='.6' fill='%23dc3545' stroke='none'/%3e%3c/svg%3e");
    background-repeat:no-repeat;
    background-position:right calc(0.375em + 0.1875rem) center;
    background-size:calc(0.75em + 0.375rem) calc(0.75em + 0.375rem)
}
.Content-Picker-Required .multiselect__select{
    padding-right:calc(1.5em + 0.95rem)
}
`
@hishamco  here it is.

hishamco commented:
So let us create a style file for it like what @Piedone suggested

Sandeep905 commented:
> So let us create a style file for it like what @Piedone suggested

are you on it @hishamco  ?

hishamco commented:
I didn't understand what you mean

Sandeep905 commented:
> I didn't understand what you mean

Are you creating the style file as suggested for the custom CSS I shared?

hishamco commented:
I mean create a SCSS file with your changes

Sandeep905 commented:
> @Sandeep905 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
> ### Contribution License Agreement
> This Contribution License Agreement ( **“Agreement”** ) is agreed to by the party signing below ( **“You”** ), and conveys certain license rights to the .NET Foundation ( **“.NET Foundation”** ) for Your contributions to .NET Foundation open source projects. This Agreement is effective as of the latest signature date below.
> 
> **1. Definitions.**
> 
> **“Code”** means the computer software code, whether in human-readable or machine-executable form, that is delivered by You to .NET Foundation under this Agreement.
> 
> **“Project”** means any of the projects owned or managed by .NET Foundation and offered under a license approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).
> 
> **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any Project, including but not limited to communication on electronic mailing lists, source code control systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of discussing and improving that Project, but excluding communication that is conspicuously marked or otherwise designated in writing by You as “Not a Submission.”
> 
> **“Submission”** means the Code and any other copyrightable material Submitted by You, including any associated comments and documentation.
> 
> **2. Your Submission.** You must agree to the terms of this Agreement before making a Submission to any Project. This Agreement covers any and all Submissions that You, now or in the future (except as described in Section 4 below), Submit to any Project.
> 
> **3. Originality of Work.** You represent that each of Your Submissions is entirely Your original work. Should You wish to Submit materials that are not Your original work, You may Submit them separately to the Project if You (a) retain all copyright and license information that was in the materials as you received them, (b) in the description accompanying your Submission, include the phrase "Submission containing materials of a third party:" followed by the names of the third party and any licenses or other restrictions of which You are aware, and (c) follow any other instructions in the Project's written guidelines concerning Submissions.
> 
> **4. Your Employer.** References to “employer” in this Agreement include Your employer or anyone else for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your Submission is made in the course of Your work for an employer or Your employer has intellectual property rights in Your Submission by contract or applicable law, You must secure permission from Your employer to make the Submission before signing this Agreement. In that case, the term “You” in this Agreement will refer to You and the employer collectively. If You change employers in the future and desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement and secure permission from the new employer before Submitting those Submissions.
> 
> **5. Licenses.**
> 
> **a. Copyright License.** You grant .NET Foundation, and those who receive the Submission directly or indirectly from .NET Foundation, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third parties.
> 
> **b. Patent License.** You grant .NET Foundation, and those who receive the Submission directly or indirectly from .NET Foundation, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under Your patent claims that are necessarily infringed by the Submission or the combination of the Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and import or otherwise dispose of the Submission alone or with the Project.
> 
> **c. Other Rights Reserved.** Each party reserves all rights not expressly granted in this Agreement. No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are granted by implication, exhaustion, estoppel or otherwise.
> 
> **6. Representations and Warranties.** You represent that You are legally entitled to grant the above licenses. You represent that each of Your Submissions is entirely Your original work (except as You may have disclosed under Section 3 ). You represent that You have secured permission from Your employer to make the Submission in cases where Your Submission is made in the course of Your work for Your employer or Your employer has intellectual property rights in Your Submission by contract or applicable law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You have the necessary authority to bind the listed employer to the obligations contained in this Agreement. You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES EXPRESSLY STATED IN SECTIONS 3, 4, AND 6 , THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
> 
> **7. Notice to .NET Foundation.** You agree to notify .NET Foundation in writing of any facts or circumstances of which You later become aware that would make Your representations in this Agreement inaccurate in any respect.
> 
> **8. Information about Submissions.** You agree that contributions to Projects and information about contributions may be maintained indefinitely and disclosed publicly, including Your name and other information that You submit with Your Submission.
> 
> **9. Governing Law/Jurisdiction.** This Agreement is governed by the laws of the State of Washington, and the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County, Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all defenses of lack of personal jurisdiction and forum non-conveniens.
> 
> **10. Entire Agreement/Assignment.** This Agreement is the entire agreement between the parties, and supersedes any and all prior agreements, understandings or communications, written or oral, between the parties relating to the subject matter hereof. This Agreement may be assigned by .NET Foundation.
> 
> _.NET Foundation dedicates this Contribution License Agreement to the public domain according to the Creative Commons CC0 1._

@dotnet-policy-service agree

Sandeep905 commented:
> > @Sandeep905 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> > ```
> > @dotnet-policy-service agree [company="{your company}"]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > > Options:
> > > 
> > > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > > 
> > > ```
> > > @dotnet-policy-service agree
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > > 
> > > ```
> > > @dotnet-policy-service agree company="Microsoft"
> > > ```
> > 
> > 
> > Contributor License Agreement
> > ### Contribution License Agreement
> > This Contribution License Agreement ( **“Agreement”** ) is agreed to by the party signing below ( **“You”** ), and conveys certain license rights to the .NET Foundation ( **“.NET Foundation”** ) for Your contributions to .NET Foundation open source projects. This Agreement is effective as of the latest signature date below.
> > **1. Definitions.**
> > **“Code”** means the computer software code, whether in human-readable or machine-executable form, that is delivered by You to .NET Foundation under this Agreement.
> > **“Project”** means any of the projects owned or managed by .NET Foundation and offered under a license approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).
> > **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any Project, including but not limited to communication on electronic mailing lists, source code control systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of discussing and improving that Project, but excluding communication that is conspicuously marked or otherwise designated in writing by You as “Not a Submission.”
> > **“Submission”** means the Code and any other copyrightable material Submitted by You, including any associated comments and documentation.
> > **2. Your Submission.** You must agree to the terms of this Agreement before making a Submission to any Project. This Agreement covers any and all Submissions that You, now or in the future (except as described in Section 4 below), Submit to any Project.
> > **3. Originality of Work.** You represent that each of Your Submissions is entirely Your original work. Should You wish to Submit materials that are not Your original work, You may Submit them separately to the Project if You (a) retain all copyright and license information that was in the materials as you received them, (b) in the description accompanying your Submission, include the phrase "Submission containing materials of a third party:" followed by the names of the third party and any licenses or other restrictions of which You are aware, and (c) follow any other instructions in the Project's written guidelines concerning Submissions.
> > **4. Your Employer.** References to “employer” in this Agreement include Your employer or anyone else for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your Submission is made in the course of Your work for an employer or Your employer has intellectual property rights in Your Submission by contract or applicable law, You must secure permission from Your employer to make the Submission before signing this Agreement. In that case, the term “You” in this Agreement will refer to You and the employer collectively. If You change employers in the future and desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement and secure permission from the new employer before Submitting those Submissions.
> > **5. Licenses.**
> > **a. Copyright License.** You grant .NET Foundation, and those who receive the Submission directly or indirectly from .NET Foundation, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third parties.
> > **b. Patent License.** You grant .NET Foundation, and those who receive the Submission directly or indirectly from .NET Foundation, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under Your patent claims that are necessarily infringed by the Submission or the combination of the Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and import or otherwise dispose of the Submission alone or with the Project.
> > **c. Other Rights Reserved.** Each party reserves all rights not expressly granted in this Agreement. No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are granted by implication, exhaustion, estoppel or otherwise.
> > **6. Representations and Warranties.** You represent that You are legally entitled to grant the above licenses. You represent that each of Your Submissions is entirely Your original work (except as You may have disclosed under Section 3 ). You represent that You have secured permission from Your employer to make the Submission in cases where Your Submission is made in the course of Your work for Your employer or Your employer has intellectual property rights in Your Submission by contract or applicable law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You have the necessary authority to bind the listed employer to the obligations contained in this Agreement. You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES EXPRESSLY STATED IN SECTIONS 3, 4, AND 6 , THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
> > **7. Notice to .NET Foundation.** You agree to notify .NET Foundation in writing of any facts or circumstances of which You later become aware that would make Your representations in this Agreement inaccurate in any respect.
> > **8. Information about Submissions.** You agree that contributions to Projects and information about contributions may be maintained indefinitely and disclosed publicly, including Your name and other information that You submit with Your Submission.
> > **9. Governing Law/Jurisdiction.** This Agreement is governed by the laws of the State of Washington, and the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County, Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all defenses of lack of personal jurisdiction and forum non-conveniens.
> > **10. Entire Agreement/Assignment.** This Agreement is the entire agreement between the parties, and supersedes any and all prior agreements, understandings or communications, written or oral, between the parties relating to the subject matter hereof. This Agreement may be assigned by .NET Foundation.
> > _.NET Foundation dedicates this Contribution License Agreement to the public domain according to the Creative Commons CC0 1._
> 
> @dotnet-policy-service agree

@dotnet-policy-service agree company="Microsoft"

sebastienros commented:
@Sandeep905 maybe you can check [this pr](https://github.com/OrchardCMS/OrchardCore/pull/17443) which applies the same fix but on taxonomy editors.

Piedone commented:
It adds the exact CSS classes you have to use here too. So, good news, no need for new CSS!

Piedone commented on src/OrchardCore.Modules/OrchardCore.ContentFields/Views/ContentPickerField.Edit.cshtml:
```diff
@@ -12,6 +12,7 @@
     var searchUrl = Url.RouteUrl(new { area = "OrchardCore.ContentFields", controller = "ContentPickerAdmin", action = "SearchContentItems", part = partName, field = fieldName });
     var vueElementId = $"ContentPicker_{partName}_{fieldName}_{Guid.NewGuid().ToString("n")}";
     var multiple = settings.Multiple.ToString().ToLowerInvariant();
+    var isValidationRequired = settings.Required && !Model.SelectedItems.Any() && !ViewData.ModelState.IsValid;
```
This can also be `true` if there is any validation error, from any other field/part, including any other Content Picker Field (keep in mind that you can have multiple such fields on a content type). Better to check for the validation errors with the current prefix. Check from here for inspiration: https://github.com/OrchardCMS/OrchardCore/blob/49aa9efbeff51949ff726e615f1d6d1d4fc63871/src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/ContentPickerFieldDisplayDriver.cs#L115

Piedone reviewed (changes requested):
There's a validation error now even if you select an item in the picker.

Also, when you're done addressing all feedback of a review, click ["Re-request review"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#re-requesting-a-review) in the top-right corner for each reviewer when you're ready for another round of review, so they know that you're done.

Sandeep905 commented:
> There's a validation error now even if you select an item in the picker.
> 
> Also, when you're done addressing all feedback of a review, click ["Re-request review"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#re-requesting-a-review) in the top-right corner for each reviewer when you're ready for another round of review, so they know that you're done.

Yes, @Piedone. It breaks when the class is added to Picker. I am trying to fix it. I tried to add a do not merge tag but could not do that.

hishamco commented on src/OrchardCore.Modules/OrchardCore.ContentFields/Views/ContentPickerField.Edit.cshtml:
```diff
@@ -44,7 +44,7 @@
             <div class="w-lg-75 w-xxl-50">
                 <input asp-for="ContentItemIds" type="hidden" v-model="selectedIds" />
-                <vue-multiselect v-model="value" :options="options" track-by="id"
+                <vue-multiselect asp-validation-class-for="ContentItemIds" v-model="value" :options="options" track-by="id"
```
That's the only thing we need :)

Sandeep905 replied:
Yes, this is breaking the control as Zoltan Said.

Sandeep905 replied:
@hishamco @Piedone, I didn't use the existing CSS. I have added a custom one in the same file.

While Using the existing one, It is applying on the ` <vue-multiselect asp-validation-class-for="ContentItemIds"`. Don't know the reason, but it stops working. To resolve that issue, the custom CSS is the safest option for me.

https://github.com/user-attachments/assets/d923646d-d2ae-4892-b84c-bccc1fd4c9c8

hishamco commented:
I see there's a new change after I approved, which is makes me worry was the previous class working properly or not?!!

Piedone commented on src/OrchardCore.Themes/TheAdmin/Assets/scss/components/_validations.scss:
```diff
@@ -112,3 +112,14 @@ textarea.form-control.input-validation-error {
     border-color: var(--bs-form-invalid-border-color);
     padding: 0.375rem 0.75rem;
 }
+.has-validation-error .multiselect__tags {
+    border: 1px solid red !important;
+    background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 12 12' width='12' height='12' fill='none' stroke='%23dc3545'%3e%3ccircle cx='6' cy='6' r='4.5'/%3e%3cpath stroke-linejoin='round' d='M5.8 3.6h.4L6 6.5z'/%3e%3ccircle cx='6' cy='8.2' r='.6' fill='%23dc3545' stroke='none'/%3e%3c/svg%3e");
+    background-repeat: no-repeat;
+    background-position: right calc(0.375em + 0.1875rem) center;
+    background-size: calc(0.75em + 0.375rem) calc(0.75em + 0.375rem)
+}
+
+.has-validation-error .multiselect__select {
+    padding-right: calc(1.5em + 0.95rem)
+}
```
Don't repeat styling. Try to reuse the `input-validation-error` class on an element. Better change the HTML structure of the editor to make it compatible with it than add a custom validation error CSS just for this case.

And as mentioned previously, if new CSS is needed, you'll need to run SCSS compilation with Gulp: https://docs.orchardcore.net/en/latest/guides/gulp-pipeline/. If something is Content Picker-specific, then it also needs to be added only the `OrchardCore.ContentFields` module, and included from only this cshtml. If you search for "optionsEditor.css" in that project, you'll see an end-to-end example of this this can be done.

hishamco commented:
I think we change the assets management, @Sandeep905, please react to the @Piedone and let us finalize this PR

Sandeep905 commented:
> I think we change the assets management, @Sandeep905, please react to the @Piedone and let us finalize this PR

@hishamco  I am Unavailable For Some Time. If you like to fix this you can pick this issue.

MikeAlhayek commented:
@hishamco sounds like @Sandeep905 abandoned this PR. Are you able to take it over? Otherwise, close it.

hishamco commented:
I added this PR to my list, hope to revise the issue in the upcoming days
